crates/client/rpc/src/errors.rs

```diff
@@ -1,67 +1,139 @@
+use core::fmt;
+
+use blockifier::transaction::errors::TransactionExecutionError;
+
 use jsonrpsee::types::error::{CallError, ErrorObject};
-use pallet_starknet_runtime_api::StarknetTransactionExecutionError;
+use starknet_api::core::ContractAddress;
+use thiserror::Error;
 
 // Comes from the RPC Spec:
 // https://github.com/starkware-libs/starknet-specs/blob/0e859ff905795f789f1dfd6f7340cdaf5015acc8/api/starknet_write_api.json#L227
-#[derive(thiserror::Error, Clone, Copy, Debug)]
+#[derive(Error, Debug)]
 pub enum StarknetRpcApiError {
     #[error("Failed to write transaction")]
-    FailedToReceiveTxn = 1,
-    #[error("Contract not found")]
-    ContractNotFound = 20,
+    FailedToReceiveTxn,
+    #[error("Contract not found: {0}")]
```

Comment:
This is not coherent with the specs defined here: https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L3912C13-L3915C14